pkg/operator/operator.go

```diff
@@ -133,6 +137,34 @@ func New(ctx context.Context, config operatorconfig.Config, kubeConfig *rest.Con
 		return nil, fmt.Errorf("failed to create status controller: %v", err)
 	}
 
+	// Set up the DNSNameResolver controller. This controller is unmanaged by
+	// the manager; the enable_dnsnameresolver controller starts it and the
+	// caches.
+	dnsNameResolverController, dnsNameResolverControllerCaches, err := dnsnameresolver.NewUnmanaged(operatorManager, dnsnameresolver.Config{
+		OperandNamespace: operatorcontroller.DefaultOperandNamespace,
+		ServiceName: operatorcontroller.DNSServiceName(&operatorv1.DNS{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: operatorcontroller.DefaultDNSController,
+			},
+		}).Name,
+		DNSPort:                  fmt.Sprint(manifests.DNSDaemonSet().Spec.Template.Spec.Containers[0].Ports[0].ContainerPort),
```

Comment:
What does it use the DNSPort for?  Why is it only taking a single value?

Reply:
DNSPort is used for taking port on which CoreDNS pods are listening as an input. Previously it was hardcoded in the dnsnameresolver controller which was causing it to fail.